app/controllers/samples_controller.rb

```diff
@@ -190,46 +190,8 @@ def bulk_destroy
     redirect_to samples_path, notice: 'Samples were successfully deleted.'
   end
 
-  def transfer
-    new_owner = Institution.find_by(uuid: params["institution_id"])
-    if new_owner.nil?
-      flash[:notice] = "Destination Institution does not exists"
-    else
-      begin
-        change_ownership(new_owner)
-        flash[:notice] = "All samples has been transferred successfully."
-      rescue
-        flash[:notice] = "Samples transfer failed."
-      end
-    end
-
-    render json: {status: :ok}
-  end
-
   private
 
-  def change_ownership(new_owner)
-    samples = Sample.joins(:sample_identifiers).where("sample_identifiers.uuid": params["samples"])
-    Sample.transaction do
-      samples.each do |to_transfer|
-        raise "User not authorized for transferring Samples " unless authorize_resource?(to_transfer, UPDATE_SAMPLE)
-
-        transfer = SampleTransfer.new(
-          sample: to_transfer,
-          receiver_institution: new_owner
-        )
-        transfer.confirm
-        transfer.save!
-
-        if to_transfer.batch_id.nil?
-          to_transfer.site_id = nil
-          to_transfer.institution = new_owner
-          to_transfer.save!
-        end
```

Comment:
You no longer check for `batch_id.nil?` when transferring samples when confirming reception. Is it on purpose?

Reply:
Yes, good catch. I wanted to add a comment about that but forgot.
I intentionally omitted that because it's going to be removed in https://github.com/instedd/cdx/issues/1388. I suppose we should merge these PRs in order, so this PR will need an update anyways.